### ISSUE TITLE
Strict pub/sub is not that hard

### ISSUE THREAD

**Issue by alexhultman** [uNetworking/uWebSockets#1067]
It's actually not that hard to make the pub/sub optimizations stricter in terms of subscribe/unsubscribe vs. publish:

* When a socket subscribes to a topic which is triggered, simply drain before subscribing
* When unsubscribing from a topic which is triggered, simply drain before subscribing

That's all. Very simple. This way there's perfect "sync" between subscribe, unsubscribe and publish.

This is not a priority, but if someone finds this an issue it could be implemented.

**Comment by hst-m** [comment]
That makes sense and seems to be the way it should work. I think on this line you meant "unsubscribing" 
> When unsubscribing from a topic which is triggered, simply drain before **subscribing**

**Comment by alexhultman-2fa** [comment]
Yeah this change would be like 2 lines of code added. It makes a lot of sense to have them strict. It doesn't affect performance in common usage while it brings guarantees that most people expect.
